app/src/main/java/com/example/habitsmasher/ui/dashboard/AddHabitDialog.java

```diff
@@ -0,0 +1,134 @@
+            @Override
+            public void onClick(View view) {
+                Log.d(TAG, "Cancel");
+                getDialog().dismiss();
+            }
+        });
+
+        _confirmNewHabit.setOnClickListener(new View.OnClickListener() {
+            @RequiresApi(api = Build.VERSION_CODES.O)
+            @Override
+            public void onClick(View view) {
+                Log.d(TAG, "Confirm");
+                String habitTitleInput = _habitTitleEditText.getText().toString();
+                String habitReasonInput = _habitReasonEditText.getText().toString();
+                String habitDateInput = _habitDateEditText.getText().toString();
+                DateFormat inputDateFormatter = new SimpleDateFormat(DATE_FORMAT);
+                _invalidDate = false;
+                Date habitDate = null;
+                try {
+                    inputDateFormatter.setLenient(false);
+                    habitDate = inputDateFormatter.parse(habitDateInput);
+                } catch (ParseException e) {
+                    _invalidDate = true;
+                    e.printStackTrace();
+                }
+
+                /**
+                 * When user input meets the requirements for a proper habit title, habit reason, and habit date,
+                 * have the habit be added to the habit list back in dashboard fragment.
+                 */
+                if (((habitTitleInput.length()>0)&&(habitTitleInput.length()<=20))&&
+                        ((habitReasonInput.length()>0)&&(habitReasonInput.length()<=30))&&
+                        (!(habitDateInput.equals("")))&&
+                        (!(_invalidDate))
+                ) {
+                    _listener.addNewHabit(habitTitleInput, habitReasonInput, habitDate);
+                    _listener.addNewHabitFirebase(habitTitleInput, habitReasonInput, habitDate);
+                    getDialog().dismiss();
+                } else{
+
+                    //Handle error checking for new habit name/title
+                    if ((!(habitTitleInput.length()>0)) || (!(habitTitleInput.length()<=20))){
+                        Toast.makeText(getActivity(), "Incorrect habit title entered", Toast.LENGTH_SHORT).show();
+                    }
+
+                    //Handle error checking for new habit reason
+                    if ((!(habitReasonInput.length()>0)) || (!(habitReasonInput.length()<=30))){
+                        Toast.makeText(getActivity(), "Incorrect habit reason entered", Toast.LENGTH_SHORT).show();
+                    }
+
+                    //Handle error checking for new habit date.
+                    if ((habitDateInput.equals(""))){
+                        Toast.makeText(getActivity(), "Please enter a start date", Toast.LENGTH_SHORT).show();
+                    } else if(_invalidDate==true){
+                        Toast.makeText(getActivity(), "Habit start date format: dd-mm-yyyy", Toast.LENGTH_SHORT).show();
```

Comment:
Minor: Another suggestion for a follow-up PR, each "error message" could be extracted out to it's own **constant**.